src/request-pipeline/destination-request/http2.ts

```diff
@@ -0,0 +1,137 @@
+import http2, { ClientHttp2Session, Http2Stream, IncomingHttpHeaders, IncomingHttpStatusHeader } from 'http2';
```

Comment:
This file is large. I propose to split its content to few files
```js
http2/sessions.ts
http2/format-request-headers.ts
http2/create-response-like.ts
```

Also I propose to rename `makePseudoResponse` function to `createResponseLike`.